dashboard/modules/job/tests/test_job_agent.py

```diff
@@ -314,5 +317,166 @@ async def test_runtime_env_setup_failure(job_sdk_client):
+    client = JobAgentSubmissionClient(format_web_url(agent_address))
+
+    def _check_nodes():
+        try:
+            response = requests.get(webui_url + "/nodes?view=summary")
+            response.raise_for_status()
+            summary = response.json()
+            assert summary["result"] is True, summary["msg"]
+            summary = summary["data"]["summary"]
+            assert len(summary) == 3
+            for node_info in summary:
+                node_id = node_info["raylet"]["nodeId"]
+                response = requests.get(webui_url + f"/nodes/{node_id}")
+                response.raise_for_status()
+                detail = response.json()
+                assert detail["result"] is True, detail["msg"]
+                detail = detail["data"]["detail"]
+                assert detail["raylet"]["state"] == "ALIVE"
+            response = requests.get(webui_url + "/test/dump?key=agents")
+            response.raise_for_status()
+            agents = response.json()
+            assert len(agents["data"]["agents"]) == 3
+            return True
+        except Exception as ex:
+            logger.info(ex)
+            return False
+
+    wait_for_condition(_check_nodes, timeout=15)
+
+    job_ids = []
+    job_check_status = []
+    JOB_NUM = 10
+    job_agent_ports = [
+        DEFAULT_DASHBOARD_AGENT_LISTEN_PORT,
+        DEFAULT_DASHBOARD_AGENT_LISTEN_PORT + 1,
+        DEFAULT_DASHBOARD_AGENT_LISTEN_PORT + 2,
+    ]
+    for index in range(JOB_NUM):
+        runtime_env = RuntimeEnv().to_dict()
+        request = validate_request_type(
+            {
+                "runtime_env": runtime_env,
+                "entrypoint": f"while true; do echo hello index-{index}"
+                " && sleep 3600; done",
+            },
+            JobSubmitRequest,
+        )
+
+        submit_result = await client.submit_job_internal(request)
+        job_ids.append(submit_result.submission_id)
+        job_check_status.append(False)
+
+    async def _check_all_jobs_log():
+        response = requests.get(webui_url + "/nodes?view=summary")
+        response.raise_for_status()
+        summary = response.json()
+        assert summary["result"] is True, summary["msg"]
+        summary = summary["data"]["summary"]
+
+        for index, job_id in enumerate(job_ids):
+            if job_check_status[index]:
+                continue
+            result_log = f"hello index-{index}"
+            gcs_aio_client = GcsAioClient(
+                address=cluster.address, nums_reconnect_retry=0
+            )
+            supervisor_actor_info = await get_supervisor_actor_into(
+                gcs_aio_client, job_id
+            )
+
+            node_id = supervisor_actor_info.actor_table_data.address.raylet_id.hex()
+            for node_info in summary:
+                if node_info["raylet"]["nodeId"] == node_id:
+                    break
+            assert node_info["raylet"]["nodeId"] == node_id, f"node id: {node_id}"
+
```

Comment:
It is a complex test, I submit 10 jobs to a JobAgent, and those job drivers could run in any node(it depends where those SupervisorActor launched), and I will try to get those drivers' logs.

The reason why the test is so complicated is because the SuperviosrActor info has its running node IP, but it is very troublesome to obtain the HTTP port of the agent through this node IP, but it will be very simple to do this in the head node in the end, because DataSources.nodes has all the needed information